src/runtime/cuda/cuda_hardware_manager.cpp

```diff
@@ -172,6 +172,15 @@ cuda_hardware_context::get_property(device_uint_property prop) const {
   case device_uint_property::max_group_size:
     return _properties->maxThreadsPerBlock;
     break;
+  case device_uint_property::max_num_sub_groups:
+    return _properties->maxThreadsPerBlock / _properties->warpSize;
+    break;
+  case device_uint_property::sub_group_independent_forward_progress:
+    return (_properties->major >= 7) ? 1 : 0; // True on Volta or newer
```

Comment:
The spec says:
> Returns true if the device supports independent forward progress of sub-groups with respect to other sub-groups in the same work-group.

So this is about forward progress of *different* subgroups/warps with respect to each other. My understanding is that this was also true before Volta. I think what has changed with Volta is forward-progress of individual threads within a single warp.